packages/material-ui/src/styles/createPalette.d.ts

```diff
@@ -65,10 +65,10 @@ export interface Palette {
   background: TypeBackground;
   getContrastText: (background: string) => string;
   augmentColor: (
-    color: SimplePaletteColorOptions,
-    mainShade: number | string,
-    lightShade: number | string,
-    darkShade: number | string,
+    color: PaletteColorOptions,
```

Comment:
That's not completely correct. See https://github.com/mui-org/material-ui/blob/db584f0d966ab49b89e941058a1a7ee8a96ad8b7/packages/material-ui/src/styles/createPalette.js#L133

You would need add some overload so that you require a `mainShade` if the first argument is `Partial<Color>` and only make `mainShade` optional if `SimplePaletteColorOptions` is provided.

Reply:
It looks correct to me 🤔.

Reply:
You're right. If it's undefined then the `mainShade` is 500 and `color.main` will be set to `color[500]` (which we should probably rewrite. We're mutating that object and I think it's not that far fetched that someone iterates over the properties i.e. listing all shades).

Would still like some tests for it though.

Reply:
> e're mutating that object

We have #12390 for that 😆

Reply:
Oh now I get it. I was so confused by that title. It should say pure not immutable.